src/control/lib/control/pool.go

```diff
@@ -183,7 +183,8 @@ func (r *poolRequest) getDeadline() time.Time {
 	if !r.deadline.IsZero() {
 		return r.deadline
 	}
-	return time.Now().Add(DefaultPoolTimeout)
+	r.SetTimeout(DefaultPoolTimeout)
```

Comment:
From what I understand this is a fix not related to not removing the pool label when pool destroy fail ?

From my side, if possible, I would prefer to set this timeout default value at the initialization of the poolRequest instead of as a side effect of a getter.

Reply:
OKn thanks @mjmac for the explanation.